UpsertCgnStatus/handler.ts

```diff
@@ -38,55 +48,72 @@ import { checkUpdateCgnIsRunning } from "../utils/orchestrators";
 type ErrorTypes =
   | IResponseErrorInternal
   | IResponseErrorNotFound
+  | IResponseErrorForbiddenNotAuthorized
   | IResponseErrorConflict;
 type ReturnTypes =
   | IResponseSuccessAccepted
   | IResponseSuccessRedirectToResource<InstanceId, InstanceId>
   | ErrorTypes;
 
-type IRevokeCgnHandler = (
+type IUpsertCgnStatusHandler = (
   context: Context,
   fiscalCode: FiscalCode,
-  CgnRevocationRequest: CgnRevocationRequest
+  cgnStatusUpsertRequest: CgnStatusUpsertRequest
 ) => Promise<ReturnTypes>;
 
-export function RevokeCgnHandler(
+const toCgnStatus = (cgnStatusUpsertRequest: CgnStatusUpsertRequest) => {
+  return cgnStatusUpsertRequest.action === ActionEnum.REVOKE
+    ? {
+        revocation_date: new Date(),
+        revocation_reason: cgnStatusUpsertRequest.revocation_reason,
+        status: StatusEnum.REVOKED
+      }
+    : {
+        status: PendingStatusEnum.PENDING
+      };
+};
+
+export function UpsertCgnStatusHandler(
   userCgnModel: UserCgnModel,
-  logPrefix: string = "RevokeCgnHandler"
-): IRevokeCgnHandler {
-  return async (context, fiscalCode, revocationReq) => {
+  logPrefix: string = "UpsertCgnHandler"
```

Comment:
```suggestion
  logPrefix: string = "UpsertCgnStatusHandler"
```

Reply:
Fixed in [0f2f273](https://github.com/pagopa/io-functions-cgn/pull/12/commits/0f2f273bd086ff2a263a5e50104cb22cbcb96cc4)